ibm/service/vpc/data_source_ibm_is_bare_metal_server.go

```diff
@@ -818,6 +823,10 @@ func dataSourceIBMISBareMetalServerRead(context context.Context, d *schema.Resou
 		return diag.FromErr(fmt.Errorf("[ERROR] Error setting identifier: %s", err))
 	}
 
+	if err = d.Set(isBareMetalServerFirmwareUpdateTypeAvailable, *bms.Firmware.Update); err != nil {
```

Comment:
nil check on `Firmware` needed

Reply:
`Firmware` is a required field in response as well as `Update`. Hence I didn't add. Do you think it can be nil ?